_posts/2019-03-08-code-quality-inspection-for-clojure-with-sonarqube.md

```diff
@@ -0,0 +1,194 @@
+---
+layout: post
+title: Code quality inspection for Clojure using SonarQube  
+author: hjhamala
+excerpt: Clojure has good libraries for code quality analyzing and vulnerabilities detection. I introduce the support for Kibit, ancient-clj, lein-nvd and Cloverage in existing Clojure SonarQube plugin. 
+tags:
+- SonarQube
+- Clojure
+- ClojureScript
+- Code quality
+- Vulnerabilities detection
+- Continuous inspection
+---
+Building good and secure code has never been easy. Programmers make mistakes during development. Programs use external libraries which may have  bugs and vulnerabilities. New vulnerabilities are found which forces updates to existing programs.
+
+Software development has many ways to battle against bugs. For example testing, code reviews and pair programming can help us to make better and safer programs. Fortunately we may also use code analyzers which automate the testing and vulnerability finding.
+
+[**Clojure**](https://clojure.org/) is not immune to bugs or vulnerabilities. Clojure being a dynamic language leads to situations where some problems may occur at runtime which in static languages are found out at compile time. The Clojure community has produced many useful tools such as linters and security scanners.
+
+## Clojure code quality tools
+
+### Eastwood
+[Eastwood](https://github.com/jonase/eastwood) is a lint tool for Clojure (no CLJS support) which detects many kinds of potentially problematic code. 
+
+For example the following Clojure code contains a potential bug:
```

Comment:
For an example

Reply:
https://ell.stackexchange.com/questions/19923/for-example-vs-for-an-example